File: OldProject/SpaceFist/SpaceFist/AI/Abstract/EnemyAI.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SpaceFist.State.Abstract;
using SpaceFist.Entities;

namespace SpaceFist.AI
{
    /// <summary>
    /// The interface implemented by all enemy AI's.
    /// </summary>
    public interface EnemyAI
    {
        ShipEnemyInfo ShipEnemyInfo { get; set; }
        ShipInfo      ShipInfo      { get; set; }

        void Update();
    }
}

File: core/src/com/spacefist/ai/abst/EnemyAI.java
package com.spacefist.ai.abst;


import com.spacefist.ai.ShipEnemyInfo;
import com.spacefist.ai.ShipInfo;

/**
 * The interface implemented by all enemy AI's.
 */
public interface EnemyAI {
    ShipEnemyInfo getShipEnemyInfo();
    void setShipEnemyInfo(ShipEnemyInfo shipEnemyInfo);
    ShipInfo getShipInfo();
    void setShipInfo(ShipInfo shipInfo);
    void update();
}
